includes/discovery/route.inc.php

```diff
@@ -64,7 +65,7 @@
 //if the device does not support IP-FORWARD-MIB, we can still discover the ipv4 (only)
 //routes using RFC1213 but no way to limit the amount of routes here !!
 
-if (! isset($ipForwardNb['0']['inetCidrRouteNumber'])) {
```

Comment:
Not a fan of this change, if it supports RFC1213, can't we use this and remove the custom discovery instead?

Reply:
hi @Jellyfrog 
custom discovery is only for IPv6 !!!  not IPv4
and this line
if (! isset($ipForwardNb['0']['inetCidrRouteNumber']) && $device['os'] != 'routeros') 
avoid duplicate IPv4 discovery

Reply:
> Not a fan of this change, if it supports RFC1213, can't we use this and remove the custom discovery instead?

if you be kind to look at picture attached above, snapshot from database, you will see that there is DUPLICATE ipv4 route discovery, with ccitt.0 and dotZeroDot atributes
so, ROS is discovered TWICE in old route.inc.php code
that is what i try to avoid with != routeros to skip duplicates

Reply:
Yes I understand, but where does the duplicates come from then? feels like a more generic fix would be better

Reply:
> Yes I understand, but where does the duplicates come from then? feels like a more generic fix would be better

Hi @Jellyfrog 
it is discovered twice because MT dispose both
RFC1213 && IP-FORWARD-MIB

on first, it is discovered as ccitt.0, on second, it is discovered as zeroDotZero

here is the output with DISABLED RFC1213 as in PR

SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'COMMUNITY' '-OQUsetX' '-m' 'IP-FORWARD-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/mikrotik' 'udp6:HOSTNAME:255::1]:161' '.1.3.6.1.2.1.4.24.4.1']

Exitcode: 1  

Error: OID not increasing: ipCidrRouteDest[169.254.106.0][255.255.255.0][0][169.254.255.206]
 >= ipCidrRouteDest[169.254.106.0][255.255.255.0][0][169.254.255.206]  

ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *
ipCidrRouteDest[*][*][0][*] = *  

Error: OID not increasing: ipCidrRouteDest[169.254.106.0][255.255.255.0][0][169.254.255.206]
 >= ipCidrRouteDest[169.254.106.0][255.255.255.0][0][169.254.255.206]

Reply:
zeroDotZero and ccitt.0 are both 0.0, just with different mibs.

perhaps we can tell it to return numeric data instead to prevent duplicates.

Reply:
> zeroDotZero and ccitt.0 are both 0.0, just with different mibs.
> 
> perhaps we can tell it to return numeric data instead to prevent duplicates.

i don't feel brave enough to make huge change in route.inc.php
as you could see, i choose pCidrRouteTable to "modernize" with SnmpQuery because of low impact, only few simdata was changed. Don't have enough fantasy to predict what will happen if i mess with rest of route.inc.php

Reply:
@Npeca75 you do pretty well ;)  I think it is just these if os checks tend to bite us later.